Unable to install Strimzi operator on microshift

I am unable to install Strimzi operator on Microshift I am getting the following error.

```
2021-07-21 15:38:45 INFO  Main:61 - ClusterOperator 0.24.0 is starting
2021-07-21 15:38:46 INFO  Main:63 - Cluster Operator configuration is ClusterOperatorConfig(namespaces=[*],reconciliationIntervalMs=120000,operationTimeoutMs=300000,connectBuildTimeoutMs=300000,createClusterRoles=false,versions=versions{2.7.0={proto: 2.7 msg: 2.7 kafka-image: quay.io/strimzi/kafka@sha256:95cfe9000afda2f7def269ca46472d3803ee85146c521884884d8505a7187daf connect-image: quay.io/strimzi/kafka@sha256:95cfe9000afda2f7def269ca46472d3803ee85146c521884884d8505a7187daf connects2i-image: quay.io/strimzi/kafka@sha256:95cfe9000afda2f7def269ca46472d3803ee85146c521884884d8505a7187daf mirrormaker-image: quay.io/strimzi/kafka@sha256:95cfe9000afda2f7def269ca46472d3803ee85146c521884884d8505a7187daf mirrormaker2-image: quay.io/strimzi/kafka@sha256:95cfe9000afda2f7def269ca46472d3803ee85146c521884884d8505a7187daf}, 2.7.1={proto: 2.7 msg: 2.7 kafka-image: quay.io/strimzi/kafka@sha256:8959b7968ab8b3306906cdbff2ebb8d63329af37e58124a601843795c4ef5bd6 connect-image: quay.io/strimzi/kafka@sha256:8959b7968ab8b3306906cdbff2ebb8d63329af37e58124a601843795c4ef5bd6 connects2i-image: quay.io/strimzi/kafka@sha256:8959b7968ab8b3306906cdbff2ebb8d63329af37e58124a601843795c4ef5bd6 mirrormaker-image: quay.io/strimzi/kafka@sha256:8959b7968ab8b3306906cdbff2ebb8d63329af37e58124a601843795c4ef5bd6 mirrormaker2-image: quay.io/strimzi/kafka@sha256:8959b7968ab8b3306906cdbff2ebb8d63329af37e58124a601843795c4ef5bd6}, 2.8.0={proto: 2.8 msg: 2.8 kafka-image: quay.io/strimzi/kafka@sha256:fbb08410d9595029bc4a02ed859971264e6ce2dc85dd6a9855eaa7bb58b52a25 connect-image: quay.io/strimzi/kafka@sha256:fbb08410d9595029bc4a02ed859971264e6ce2dc85dd6a9855eaa7bb58b52a25 connects2i-image: quay.io/strimzi/kafka@sha256:fbb08410d9595029bc4a02ed859971264e6ce2dc85dd6a9855eaa7bb58b52a25 mirrormaker-image: quay.io/strimzi/kafka@sha256:fbb08410d9595029bc4a02ed859971264e6ce2dc85dd6a9855eaa7bb58b52a25 mirrormaker2-image: quay.io/strimzi/kafka@sha256:fbb08410d9595029bc4a02ed859971264e6ce2dc85dd6a9855eaa7bb58b52a25}},imagePullPolicy=null,imagePullSecrets=null,operatorNamespace=operators,operatorNamespaceLabels=null,rbacScope=CLUSTER,customResourceSelector=null,featureGates=FeatureGates(controlPlaneListener=false,ServiceAccountPatching=false))
2021-07-21 15:38:47 ERROR PlatformFeaturesAvailability:152 - Detection of Kubernetes version failed.
io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:64) ~[io.fabric8.kubernetes-client-5.4.1.jar:?]
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:53) ~[io.fabric8.kubernetes-client-5.4.1.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.ClusterOperationsImpl.fetchVersion(ClusterOperationsImpl.java:57) ~[io.fabric8.kubernetes-client-5.4.1.jar:?]
	at io.fabric8.kubernetes.client.DefaultKubernetesClient.getVersion(DefaultKubernetesClient.java:501) ~[io.fabric8.kubernetes-client-5.4.1.jar:?]
	at io.fabric8.kubernetes.client.DefaultKubernetesClient.getVersion(DefaultKubernetesClient.java:496) ~[io.fabric8.kubernetes-client-5.4.1.jar:?]
	at io.strimzi.operator.PlatformFeaturesAvailability.lambda$getVersionInfoFromKubernetes$5(PlatformFeaturesAvailability.java:150) ~[io.strimzi.operator-common-0.24.0.jar:0.24.0]
	at io.vertx.core.impl.ContextImpl.lambda$null$0(ContextImpl.java:160) ~[io.vertx.vertx-core-4.1.0.jar:4.1.0]
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:96) ~[io.vertx.vertx-core-4.1.0.jar:4.1.0]
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:158) ~[io.vertx.vertx-core-4.1.0.jar:4.1.0]
	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76) ~[io.vertx.vertx-core-4.1.0.jar:4.1.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [io.netty.netty-common-4.1.65.Final.jar:4.1.65.Final]
	at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: java.text.ParseException: Unparseable date: "2021-07-07-002919"
	at java.text.DateFormat.parse(DateFormat.java:395) ~[?:?]
	at io.fabric8.kubernetes.client.VersionInfo$Builder.withBuildDate(VersionInfo.java:106) ~[io.fabric8.kubernetes-client-5.4.1.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.ClusterOperationsImpl.fetchVersionInfoFromResponse(ClusterOperationsImpl.java:73) ~[io.fabric8.kubernetes-client-5.4.1.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.ClusterOperationsImpl.fetchVersion(ClusterOperationsImpl.java:55) ~[io.fabric8.kubernetes-client-5.4.1.jar:?]
	... 11 more
2021-07-21 15:38:47 ERROR Main:91 - Failed to gather environment facts
io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:64) ~[io.fabric8.kubernetes-client-5.4.1.jar:?]
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:53) ~[io.fabric8.kubernetes-client-5.4.1.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.ClusterOperationsImpl.fetchVersion(ClusterOperationsImpl.java:57) ~[io.fabric8.kubernetes-client-5.4.1.jar:?]
	at io.fabric8.kubernetes.client.DefaultKubernetesClient.getVersion(DefaultKubernetesClient.java:501) ~[io.fabric8.kubernetes-client-5.4.1.jar:?]
	at io.fabric8.kubernetes.client.DefaultKubernetesClient.getVersion(DefaultKubernetesClient.java:496) ~[io.fabric8.kubernetes-client-5.4.1.jar:?]
	at io.strimzi.operator.PlatformFeaturesAvailability.lambda$getVersionInfoFromKubernetes$5(PlatformFeaturesAvailability.java:150) ~[io.strimzi.operator-common-0.24.0.jar:0.24.0]
	at io.vertx.core.impl.ContextImpl.lambda$null$0(ContextImpl.java:160) ~[io.vertx.vertx-core-4.1.0.jar:4.1.0]
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:96) ~[io.vertx.vertx-core-4.1.0.jar:4.1.0]
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:158) ~[io.vertx.vertx-core-4.1.0.jar:4.1.0]
	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76) ~[io.vertx.vertx-core-4.1.0.jar:4.1.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [io.netty.netty-common-4.1.65.Final.jar:4.1.65.Final]
	at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: java.text.ParseException: Unparseable date: "2021-07-07-002919"
	at java.text.DateFormat.parse(DateFormat.java:395) ~[?:?]
	at io.fabric8.kubernetes.client.VersionInfo$Builder.withBuildDate(VersionInfo.java:106) ~[io.fabric8.kubernetes-client-5.4.1.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.ClusterOperationsImpl.fetchVersionInfoFromResponse(ClusterOperationsImpl.java:73) ~[io.fabric8.kubernetes-client-5.4.1.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.ClusterOperationsImpl.fetchVersion(ClusterOperationsImpl.java:55) ~[io.fabric8.kubernetes-client-5.4.1.jar:?]
	... 11 more
```

### The steps I took to deploy strimzi are below.

**Confirm that operatorhub is installed**
```
$ kubectl get pods -n olm
NAME                                READY   STATUS    RESTARTS   AGE
catalog-operator-64bd4f69f6-dwqr9   1/1     Running   0          2m43s
olm-operator-789475dcf9-jft2z       1/1     Running   0          2m43s
operatorhubio-catalog-pdpxl         1/1     Running   0          2m21s
packageserver-759454b88b-ff47x      1/1     Running   0          2m21s
packageserver-759454b88b-xdsd8      1/1     Running   0          2m21s
```

**Install the kafka operator** 
```
kubectl create -f https://operatorhub.io/install/strimzi-kafka-operator.yaml
kubectl get csv -n operators -w
```

--- a/Makefile
+++ b/Makefile
@@ -5,11 +5,13 @@ include ./vendor/github.com/openshift/build-machinery-go/make/targets/openshift/
 # TIMESTAMP is defined here, and only here, and propagated through out the build flow.  This ensures that every artifact
 # (binary version and image tag) all have the exact same build timestamp.  Because kubectl/oc expect
 # a timestamp composed with ':'s we must replace the chars with '-' so that it is still compliant with image tag format.
-export BIN_TIMESTAMP :=$(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
-export TIMESTAMP :=$(shell echo $(BIN_TIMESTAMP) | tr ':' '-')
+export BIN_TIMESTAMP ?=$(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
+export TIMESTAMP ?=$(shell echo $(BIN_TIMESTAMP) | tr ':' '-')
 
 RELEASE_PRE :=0.4.7-0.microshift
-export SOURCE_GIT_TAG :=$(shell echo $(RELEASE_PRE)-$(BIN_TIMESTAMP))
+# Overload SOURCE_GIT_TAG value set in vendor/github.com/openshift/build-machinery-go/make/lib/golang.mk
+# because since it doesn't work with our version scheme.
+SOURCE_GIT_TAG :=$(shell git describe --long --tags --abbrev=7 --broke --match '$(RELEASE_PRE)*' || echo 'v0.0.0-unknown')
 
 SRC_ROOT :=$(shell pwd)
 
@@ -21,7 +23,7 @@ CROSS_BUILD_BINDIR :=$(OUTPUT_DIR)/bin
 CTR_CMD :=$(or $(shell which podman 2>/dev/null), $(shell which docker 2>/dev/null))
 
 GO_EXT_LD_FLAGS :=-extldflags '-static'
-GO_LD_EXTRAFLAGS :=-X k8s.io/component-base/version.gitMajor=1 \
+GO_LD_FLAGS :=-ldflags "-X k8s.io/component-base/version.gitMajor=1 \
                    -X k8s.io/component-base/version.gitMinor=20 \
                    -X k8s.io/component-base/version.gitVersion=v1.20.1 \
                    -X k8s.io/component-base/version.gitCommit=5feb30e1bd3620 \
@@ -33,11 +35,15 @@ GO_LD_EXTRAFLAGS :=-X k8s.io/component-base/version.gitMajor=1 \
                    -X k8s.io/client-go/pkg/version.gitCommit=5feb30e1bd3620 \
                    -X k8s.io/client-go/pkg/version.gitTreeState=clean \
                    -X k8s.io/client-go/pkg/version.buildDate=$(BIN_TIMESTAMP) \
+                   -X github.com/microshift/pkg/version.versionFromGit=$(SOURCE_GIT_TAG) \
+                   -X github.com/microshift/pkg/version.gitCommit=$(SOURCE_GIT_COMMIT) \
+                   -X github.com/microshift/pkg/version.gitTreeState=$(SOURCE_GIT_TREE_STATE) \
+                   -X github.com/microshift/pkg/version.buildDate=$(BIN_TIMESTAMP) \
                    $(GO_EXT_LD_FLAGS) \
-                   -s -w
+                   -s -w"
 
 debug:
-	@echo FLAGS:"$(GO_LD_EXTRAFLAGS)"
+	@echo FLAGS:"$(GO_LD_FLAGS)"
 	@echo TAG:"$(SOURCE_GIT_TAG)"
 
 # These tags make sure we can statically link and avoid shared dependencies
@@ -75,9 +81,11 @@ cross-build: cross-build-linux-amd64 cross-build-linux-arm64
 # containerized build targets #
 ###############################
 _build_containerized:
+	echo BIN_TIMESTAMP==$(BIN_TIMESTAMP)
 	$(CTR_CMD) build -t $(IMAGE_REPO):$(RELEASE_PRE)-$(TIMESTAMP)-linux-$(ARCH) \
 		-f "$(SRC_ROOT)"/images/build/Dockerfile \
 		--build-arg SOURCE_GIT_TAG=$(SOURCE_GIT_TAG) \
+		--build-arg BIN_TIMESTAMP=$(BIN_TIMESTAMP) \
 		--build-arg ARCH=$(ARCH) \
 		--build-arg MAKE_TARGET="cross-build-linux-$(ARCH)" \
 		--platform="linux/$(ARCH)" \

--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,12 @@
 include ./vendor/github.com/openshift/build-machinery-go/make/golang.mk
 include ./vendor/github.com/openshift/build-machinery-go/make/targets/openshift/deps.mk
 
-TIMESTAMP :=$(shell date -u +'%Y-%m-%d-%H%M%S')
+# TIMESTAMP is defined here, and only here, and propagated through out the build flow.  This ensures that every artifact
+# (binary version and image tag) all have the exact same build timestamp.  Because kubectl/oc expect
+# a timestamp composed with ':'s we must replace the chars with '-' so that it is still compliant with image tag format.
+BIN_TIMESTAMP :=$(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
+TIMESTAMP :=$(shell echo $(BIN_TIMESTAMP) | tr ':' '-')
+
 RELEASE_PRE :=0.4.7-0.microshift
 SOURCE_GIT_TAG :=$(shell echo $(RELEASE_PRE)-$(TIMESTAMP))
 
@@ -21,13 +26,13 @@ GO_LD_EXTRAFLAGS :=-X k8s.io/component-base/version.gitMajor=1 \
                    -X k8s.io/component-base/version.gitVersion=v1.20.1 \
                    -X k8s.io/component-base/version.gitCommit=5feb30e1bd3620 \
                    -X k8s.io/component-base/version.gitTreeState=clean \
-                   -X k8s.io/component-base/version.buildDate=$(TIMESTAMP) \
+                   -X k8s.io/component-base/version.buildDate=$(BIN_TIMESTAMP) \
                    -X k8s.io/client-go/pkg/version.gitMajor=1 \
                    -X k8s.io/client-go/pkg/version.gitMinor=20 \
                    -X k8s.io/client-go/pkg/version.gitVersion=v1.20.1 \
                    -X k8s.io/client-go/pkg/version.gitCommit=5feb30e1bd3620 \
                    -X k8s.io/client-go/pkg/version.gitTreeState=clean \
-                   -X k8s.io/client-go/pkg/version.buildDate=$(TIMESTAMP) \
+                   -X k8s.io/client-go/pkg/version.buildDate=$(BIN_TIMESTAMP) \
                    $(GO_EXT_LD_FLAGS) \
                    -s -w
 

--- a/Makefile
+++ b/Makefile
@@ -35,16 +35,17 @@ GO_LD_FLAGS :=-ldflags "-X k8s.io/component-base/version.gitMajor=1 \
                    -X k8s.io/client-go/pkg/version.gitCommit=5feb30e1bd3620 \
                    -X k8s.io/client-go/pkg/version.gitTreeState=clean \
                    -X k8s.io/client-go/pkg/version.buildDate=$(BIN_TIMESTAMP) \
-                   -X github.com/microshift/pkg/version.versionFromGit=$(SOURCE_GIT_TAG) \
-                   -X github.com/microshift/pkg/version.gitCommit=$(SOURCE_GIT_COMMIT) \
-                   -X github.com/microshift/pkg/version.gitTreeState=$(SOURCE_GIT_TREE_STATE) \
-                   -X github.com/microshift/pkg/version.buildDate=$(BIN_TIMESTAMP) \
+                   -X github.com/openshift/microshift/pkg/version.versionFromGit=$(SOURCE_GIT_TAG) \
+                   -X github.com/openshift/microshift/pkg/version.commitFromGit=$(SOURCE_GIT_COMMIT) \
+                   -X github.com/openshift/microshift/pkg/version.gitTreeState=$(SOURCE_GIT_TREE_STATE) \
+                   -X github.com/openshift/microshift/pkg/version.buildDate=$(BIN_TIMESTAMP) \
                    $(GO_EXT_LD_FLAGS) \
                    -s -w"
 
 debug:
 	@echo FLAGS:"$(GO_LD_FLAGS)"
 	@echo TAG:"$(SOURCE_GIT_TAG)"
+	@echo SOURCE_GIT_TAG:"$(SOURCE_GIT_TAG)"
 
 # These tags make sure we can statically link and avoid shared dependencies
 GO_BUILD_FLAGS :=-tags 'include_gcs include_oss containers_image_openpgp gssapi providerless netgo osusergo'

--- a/Makefile
+++ b/Makefile
@@ -5,11 +5,11 @@ include ./vendor/github.com/openshift/build-machinery-go/make/targets/openshift/
 # TIMESTAMP is defined here, and only here, and propagated through out the build flow.  This ensures that every artifact
 # (binary version and image tag) all have the exact same build timestamp.  Because kubectl/oc expect
 # a timestamp composed with ':'s we must replace the chars with '-' so that it is still compliant with image tag format.
-BIN_TIMESTAMP :=$(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
-TIMESTAMP :=$(shell echo $(BIN_TIMESTAMP) | tr ':' '-')
+export BIN_TIMESTAMP :=$(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
+export TIMESTAMP :=$(shell echo $(BIN_TIMESTAMP) | tr ':' '-')
 
 RELEASE_PRE :=0.4.7-0.microshift
-SOURCE_GIT_TAG :=$(shell echo $(RELEASE_PRE)-$(TIMESTAMP))
+export SOURCE_GIT_TAG :=$(shell echo $(RELEASE_PRE)-$(BIN_TIMESTAMP))
 
 SRC_ROOT :=$(shell pwd)
 
@@ -56,7 +56,9 @@ update: update-generated-completions
 
 _build_local:
 	@mkdir -p "$(CROSS_BUILD_BINDIR)/$(GOOS)_$(GOARCH)"
-	+@GOOS=$(GOOS) GOARCH=$(GOARCH) $(MAKE) --no-print-directory build GO_BUILD_PACKAGES:=./cmd/microshift GO_BUILD_BINDIR:=$(CROSS_BUILD_BINDIR)/$(GOOS)_$(GOARCH)
+	+@GOOS=$(GOOS) GOARCH=$(GOARCH) $(MAKE) --no-print-directory build \
+		GO_BUILD_PACKAGES:=./cmd/microshift \
+		GO_BUILD_BINDIR:=$(CROSS_BUILD_BINDIR)/$(GOOS)_$(GOARCH)
 
 cross-build-linux-amd64:
 	+$(MAKE) _build_local GOOS=linux GOARCH=amd64
@@ -73,7 +75,7 @@ cross-build: cross-build-linux-amd64 cross-build-linux-arm64
 # containerized build targets #
 ###############################
 _build_containerized:
-	$(CTR_CMD) build -t $(IMAGE_REPO):$(SOURCE_GIT_TAG)-linux-$(ARCH) \
+	$(CTR_CMD) build -t $(IMAGE_REPO):$(RELEASE_PRE)-$(TIMESTAMP)-linux-$(ARCH) \
 		-f "$(SRC_ROOT)"/images/build/Dockerfile \
 		--build-arg SOURCE_GIT_TAG=$(SOURCE_GIT_TAG) \
 		--build-arg ARCH=$(ARCH) \

--- a/images/build/Dockerfile
+++ b/images/build/Dockerfile
@@ -1,12 +1,9 @@
-# DIGEST is the registry.access.redhat.com/ubi8/ubi-minimal:8.4 architecture specific image to use for the run stage
-# When specifying, assign as "@sha256: ... ".  Unset by default
-ARG DIGEST
-
 # Build stage
 FROM registry.access.redhat.com/ubi8/go-toolset as builder
 
 ARG ARCH=amd64
 ARG MAKE_TARGET=cross-build-linux-$ARCH
+ARG BIN_TIMESTAMP
 ARG SOURCE_GIT_TAG
 
 USER root
@@ -21,7 +18,7 @@ RUN yum install glibc-static -y
 
 COPY . .
 
-RUN make clean $MAKE_TARGET SOURCE_GIT_TAG=$SOURCE_GIT_TAG
+RUN make clean $MAKE_TARGET SOURCE_GIT_TAG=$SOURCE_GIT_TAG BIN_TIMESTAMP=$BIN_TIMESTAMP
 
 # Run stage
 # Containerized microshift

